pageserver/src/tenant/timeline.rs

```diff
@@ -3273,19 +2927,31 @@ impl Timeline {
         //    size length. Compaction will likely create the same set of n files afterwards.
         //
         // This failpoint is a superset of both of the cases.
-        fail_point!("compact-level0-phase1-return-same", |_| {
-            println!("compact-level0-phase1-return-same"); // so that we can check if we hit the failpoint
-            Ok(CompactLevel0Phase1Result {
-                new_layers: level0_deltas
-                    .iter()
-                    .map(|x| x.clone().downcast_delta_layer().unwrap())
-                    .collect(),
-                deltas_to_compact: level0_deltas
-                    .iter()
-                    .map(|x| x.layer_desc().clone().into())
-                    .collect(),
-            })
-        });
+        if cfg!(feature = "testing") {
+            let active = (|| {
+                ::fail::fail_point!("compact-level0-phase1-return-same", |_| true);
+                false
+            })();
+
+            if active {
+                let mut new_layers = Vec::with_capacity(level0_deltas.len());
+                for delta in &level0_deltas {
+                    // we are just faking these layers as being produced again for this failpoint
+                    new_layers.push(
+                        delta
+                            .download_and_keep_resident()
+                            .await
+                            .context("download layer for failpoint")?,
+                    );
+                }
+                // FIXME: println
+                println!("compact-level0-phase1-return-same"); // so that we can check if we hit the failpoint
```

Comment:
Why use println instead of logging?
For this purpose, we use logging in a couple of other places / did so in the past.

Reply:
This is existing code not added by me, but I did change around this.

But I guess this was simple enough failpoint problem to be just fixed.

Reply:
Will push 5da7da50d after tests finish.